### PR TITLE
Filipe/test run

### PR DESCRIPTION
**Pull request by mangas**
Fixes issues found during substreams test run.

1. Start substreams networks
2. Make BlockStreamBuilder async for build_firehose
3. Add manifest handling on substreams block  stream creation
4. Update DataSource to resolve substreams spkg
5. Replace substreams block EntityChanges, substreams mapper works over prost::Message
6. Hardcode entity name case and field name case for substreams

### CODE REVIEW

**Review by Eduard-Voiculescu — APPROVED** [review]

File: chain/substreams/src/data_source.rs [hunk 1]
```diff
@@ -48,8 +46,16 @@ impl blockchain::DataSource<Chain> for DataSource {
     }
 
     fn start_block(&self) -> BlockNumber {
-        // TODO(filipe): Figure out how to handle this
-        0
+        match self.source.package.modules {
+            Some(ref modules) => modules
+                .modules
+                .iter()
+                .min_by(|x, y| x.initial_block.cmp(&y.initial_block))
```
Comment (leoyvens) [inline chain/substreams/src/data_source.rs:53]: Try using `min_by_key` here.

File: chain/substreams/src/data_source.rs [hunk 2]
```diff
@@ -48,8 +46,16 @@ impl blockchain::DataSource<Chain> for DataSource {
...
+                .map(|m| m.initial_block.try_into().map_err(anyhow::Error::from))
+                .unwrap_or(Ok(0))
```
Comment (leoyvens) [inline chain/substreams/src/data_source.rs:55]: Not great that errors are ignored here, perhaps this start block value should be extracted and stored when the data source is resolved?

File: node/src/main.rs [hunk 3]
```diff
@@ -790,6 +822,57 @@ fn near_networks_as_chains(
     HashMap::from_iter(chains)
 }
 
+/// Return the hashmap of NEAR chains and also add them to `blockchain_map`.
```
Comment (leoyvens) [inline node/src/main.rs:825]: This comment should not refer to NEAR.

**Review by leoyvens — CHANGES_REQUESTED** [review]

File: chain/substreams/src/data_source.rs (hunk 2 above)
Comment (mangas) [reply chain/substreams/src/data_source.rs:55]: Yeah it's a good idea

File: chain/substreams/src/block_stream.rs [hunk 4]
```diff
@@ -40,16 +52,32 @@ impl BlockStreamBuilderTrait<Chain> for BlockStreamBuilder {
         let logger = chain
             .logger_factory
             .subgraph_logger(&deployment)
-            .new(o!("component" => "FirehoseBlockStream"));
+            .new(o!("component" => "SubstreamsBlockStream"));
+
+        // TODO: couldn't find a good way to pass the manifest because the code that calls this in the runner doesn't have the manifest
+        // either, as far as I can tell so this is the easy way but perhaps not ideal.
+        let manifest: SubgraphManifest<Chain> =
+            resolve_manifest(&logger, &deployment, &self.resolver).await?;
```
[the remark this answers is not in the document]
Comment (mangas) [reply chain/substreams/src/block_stream.rs:60]: Yep actually this is a great suggestion !

File: chain/substreams/src/data_source.rs [hunk 5]
```diff
@@ -201,6 +193,19 @@ impl blockchain::UnresolvedDataSource<Chain> for UnresolvedDataSource {
             package
         };
 
+        let initial_block: Option<u64> = match package.modules {
+            Some(ref modules) => modules
+                .modules
+                .iter()
+                .min_by_key(|x| x.initial_block)
+                .map(|x| x.initial_block),
```
Comment (leoyvens) [inline chain/substreams/src/data_source.rs:186]: Now this is just
```rust
.map(|x| x.initial_block)
.min()
```

**Review by leoyvens — APPROVED** [review]

**Pull request merged**